stix_shifter.py

```diff
@@ -1,48 +1,80 @@
-# Main module to be called
 import sys
 import importlib
+import argparse
 
-MODULES = ['qradar', 'dummy']
+MODULES = ['qradar', 'dummy', 'aql_passthrough']
 
 RESULTS = 'results'
 QUERY = 'query'
 
 
-def main():
+class StixShifter:
+    """
+    StixShifter class - implements translations of stix data
+    """
 
-    # In the case of converting a stix pattern to datasource query, arguments will take the form of...
-    # <module> <input_data_translation> <data>
-    # The module and input_translation_type will determine what module and method gets called
+    def __init__(self):
+        self.args = []
 
-    input_arguments = sys.argv[1:]
+    def main(self):
+        """
+        In the case of converting a stix pattern to datasource query, arguments will take the form of...
+        <module> <input_data_translation> <data>
+        The module and input_translation_type will determine what module and method gets called
+        """
 
-    module = input_arguments[0]
-    input_translation_type = input_arguments[1]
-    input_data = input_arguments[2]
+        # Process arguments
+        parser = argparse.ArgumentParser(description='stix_shifter')
+        subparsers = parser.add_subparsers(dest='command', help="sub-command help")
 
-    result = translate(module, input_translation_type, input_data)
-    print(result)
-    exit()
+        # translate query command
+        translation_parser = subparsers.add_parser('translate',
+                                             help='Translate a query or result set using a specific translation module')
+        translation_parser.add_argument('module', help='What translation module to use',
+                                  choices=MODULES)
+        translation_parser.add_argument('action', help='What translation action to perform', choices=[RESULTS, QUERY])
+        translation_parser.add_argument('data', help='The data to be translated', type=str)
 
+        self.args = parser.parse_args()
 
-def translate(module, translation_type, data):
-    if(module not in MODULES):
-        raise NotImplementedError
+        if self.args.command is None:
+            parser.print_help(sys.stderr)
+            sys.exit(1)
 
-    translator_module = importlib.import_module(
-        "src.modules."+module+"."+module+"_translator")
+        result = self.translate(self.args.module, self.args.action, self.args.data)
+        print(result)
+        exit(0)
 
-    interface = translator_module.Translator()
+    def translate(self, module, translation_type, data):
+        """
+        Translated queries to a specified format
+        :param module: What module to use
+        :type module: one of MODULES 'qradar', 'dummy'
```

Comment:
Should also include aql_passthrough here as well

Reply:
I added that now.